src/sagemaker/predictor.py

```diff
@@ -109,23 +110,38 @@ def _delete_endpoint_config(self):
         """Delete the Amazon SageMaker endpoint configuration
 
         """
-        endpoint_description = self.sagemaker_session.sagemaker_client.describe_endpoint(EndpointName=self.endpoint)
-        endpoint_config_name = endpoint_description['EndpointConfigName']
-        self.sagemaker_session.delete_endpoint_config(endpoint_config_name)
+        self.sagemaker_session.delete_endpoint_config(self._endpoint_config_name)
 
     def delete_endpoint(self, delete_endpoint_config=True):
-        """Delete the Amazon SageMaker endpoint and endpoint configuration backing this predictor.
+        """Delete the Amazon SageMaker endpoint backing this predictor. Also delete the endpoint configuration attached
+           to it if delete_endpoint_config is True.
 
         Args:
-            delete_endpoint_config (bool): Flag to indicate whether to delete the corresponding SageMaker endpoint
-                configuration tied to the endpoint. If False, only the endpoint will be deleted. (default: True)
+            delete_endpoint_config (bool, optional): Flag to indicate whether to delete endpoint configuration together
+                with endpoint. Defaults to True. If True, both endpoint and endpoint configuration will be deleted. If
+                False, only endpoint will be deleted.
 
         """
         if delete_endpoint_config:
             self._delete_endpoint_config()
 
         self.sagemaker_session.delete_endpoint(self.endpoint)
 
+    def delete_model(self):
+        """Deletes the Amazon SageMaker models backing this predictor.
+
+        """
+        for model_name in self._model_names:
+            self.sagemaker_session.delete_model(model_name)
+
+    def _get_model_names(self):
```

Comment:
this method does more than get the model names, it also get's the endpoint config name and sets it.